### PR TITLE
Fix error range for ETagRequired

### LINKED ISSUE #387: Fix error range for ETagRequired
When multiple child elements are missing their end tags, the error range refers to the parent element's start tag, rather than the child element's.

This error does not happen when only one child is missing their end tag.

Example XML file:
```
<?xml version="1.0" encoding="utf-8"?>
<picture>
    <p1></p1>
    <p2>    <!-- missing end tag -->
    <p3>    <!-- missing end tag -->
</picture>
```

Result:
image

image

Error message:
`The element type "p3" must be terminated by the matching end-tag "</p3>".`

### PR DESCRIPTION
**Pull request by xorye**
Fixes #387 

image

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

**Comment by fbricon** [comment]
Please use self-descriptive commit messages, like `Fix error range for ETagRequired`

**Comment by xorye** [comment]
Ok, I have just changed the commit message.

**Comment by fbricon** [comment]
LGTM but I'll let @NikolasKomonen or @angelozerr check if that's actually correct.

**Comment by NikolasKomonen** [comment]
@xorye Looks good to me

One thing to change, and I know you didn't do this. But could you update the method name of 'findChildNode' https://github.com/angelozerr/lsp4xml/blob/ccc55276ec3505f93a3d654235560f5e3504a03b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/XMLPositionUtility.java#L205

to something like 'findUnclosedChildNodeWithName'

Other than than it works great.

**Comment by fbricon** [comment]
> One thing to change, and I know you didn't do this. But could you update the method name of 'findChildNode'  to something like 'findUnclosedChildNodeWithName'

findUnclosedChildNode will be fine

**Comment by xorye** [comment]
Cool, changing it now.

**Comment by xorye** [comment]
Actually, this function finds a closed child node. Not an unclosed one.

I'll change it to findClosedChildNode

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/XMLPositionUtility.java [hunk 1]
```diff
@@ -177,20 +177,32 @@ public static Range selectChildEndTag(String childTag, int offset, DOMDocument d
...
 			}
 		}
 		return null;
 	}
 
-	static DOMNode findChildNode(String childTag, List<DOMNode> children) {
+	static DOMNode findClosedChildNode(String childTag, List<DOMNode> children) {
 		for (DOMNode child : children) {
 			if (child.isElement() && childTag != null && childTag.equals(((DOMElement) child).getTagName())
 					&& !child.isClosed()) {
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/XMLPositionUtility.java]: this child is unclosed

**Pull request merged**